src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/services_wither.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+use Symfony\Component\DependencyInjection\Argument\RewindableGenerator;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+use Symfony\Component\DependencyInjection\Exception\LogicException;
+use Symfony\Component\DependencyInjection\Exception\RuntimeException;
+use Symfony\Component\DependencyInjection\ParameterBag\FrozenParameterBag;
+
+/**
+ * This class has been auto-generated
+ * by the Symfony Dependency Injection Component.
+ *
+ * @final since Symfony 3.3
+ */
+class Symfony_DI_PhpDumper_Service_Wither extends Container
+{
+    private $parameters;
+    private $targetDirs = [];
+
+    public function __construct()
+    {
+        $this->services = $this->privates = [];
+        $this->methodMap = [
+            'wither' => 'getWitherService',
+        ];
+
+        $this->aliases = [];
+    }
+
+    public function compile()
+    {
+        throw new LogicException('You cannot compile a dumped container that was already compiled.');
+    }
+
+    public function isCompiled()
+    {
+        return true;
+    }
+
+    public function getRemovedIds()
+    {
+        return [
+            'Psr\\Container\\ContainerInterface' => true,
+            'Symfony\\Component\\DependencyInjection\\ContainerInterface' => true,
+            'Symfony\\Component\\DependencyInjection\\Tests\\Compiler\\Foo' => true,
+        ];
+    }
+
+    /**
+     * Gets the public 'wither' shared autowired service.
+     *
+     * @return \Symfony\Component\DependencyInjection\Tests\Compiler\Wither
+     */
+    protected function getWitherService()
+    {
+        $this->services['wither'] = $instance = new \Symfony\Component\DependencyInjection\Tests\Compiler\Wither();
```

Comment:
`$this->services['wither']` here looks wrong. This instance is *not* the one of the service (as the wither will replace it). But a call to get that service again somewhere during the instantiation of that service (due for instance to a circular dependency undetected at compile time because of a service breaking laziness inside its constructor, which would then have to be detected at runtime) would then return this intermediary instance as the service instance.

Reply:
Doing it this way allows each dumped line to be independent of the others. There is nothing wrong doing it this way since the registry is updated every time the instance is swapped for a new one.

Reply:
but anything *reading* the registry before the last change would read the wrong value from it. That's precisely my point.
The circular reference detection will prevent any such read from direct DIC usage during the instantiation. But it is still possible if a service calls `$container->get()` in its constructor, or fetches some of its lazy dependencies in the constructor. Both of these are considered bad practices (as they actually open the door to undetected circular references or circular references detected only at runtime in the lucky case), but they can exist. Injecting the intermediate instance would be almost impossible to debug when things go wrong because of that.

Btw, this makes me think that the new dumped code in Symfony 3.3+ which does not go through `Container::get` anymore might not detect these circular dependencies (as the protection against them is implemented in `Container::get`)